DOCS-453 add new Helm chart info (#134)
aerospike/kubernetes-operator: docs/create-cluster-helm.md

---
title: Create Aerospike Cluster Using Helm
description: How to create an Aerospike cluster using Helm
id: create-cluster-helm
---

Follow these instructions to use the Operator to deploy the Aerospike cluster using Helm.

## Requirements

Before deploying your Aerospike cluster using Helm, you must [use Helm to install the Aerospike Kubernetes Operator](install-operator-helm.md) on your cluster.

## Configure Persistent Storage

> The Aerospike Operator is designed to work with dynamically-provisioned storage classes. Aerospike Server pods may have different storage volumes associated with each service.

Reviewer: ```suggestion
The Aerospike Operator works with dynamically-provisioned storage classes. Aerospike Server pods may have different storage volumes associated with each service.
```

> Persistent storage on the pods will use these storage class provisioners to provision storage.

Reviewer: ```suggestion
Persistent storage on the pods use these storage class provisioners to provision storage.
```

Reviewer: @ebarcott-aerospike I'm just wondering if we can remove the obvious last three words in "use these storage class provisioners to provision storage." I'm actually agnostic about it, just looking to minimize words.

```shell
kubectl  -n aerospike create secret generic aerospike-secret --from-file=secrets
```

Create a Secret containing the password for the Aerospike cluster admin:

```shell
kubectl  -n aerospike create secret generic auth-secret --from-literal=password='admin123'
```

## Get the Helm Charts

To get the Helm charts, clone the `aerospike/aerospike-kubernetes-operator` repository.

```sh
git clone https://github.com/aerospike/aerospike-kubernetes-operator.git
```

The charts are in the `aerospike-kubernetes-operator/helm-charts` folder.

```sh
cd aerospike-kubernetes-operator/helm-charts
```

## Deploy the Cluster

Create a Secret containing the Aerospike feature key file `features.conf`.

```sh
kubectl create secret generic aerospike-license --from-file=[path to your features.conf file]
```

Install the chart.

```sh
helm install aerospike ./aerospike-cluster \
    --set devMode=true
```

:::tip

> This command assumes few defaults, and deploys an Aerospike cluster in "dev" mode with no data persistence. We recommend you create a custom YAML file with configurations as per your requirements and apply it with `helm install`.

Reviewer: ```suggestion
This command assumes few defaults, and deploys an Aerospike cluster in "dev" mode with no data persistence. We recommend you create a custom YAML file with your required configurations, and apply it with `helm install`.
```
